### PR TITLE
 feature: content_by_lua_file return 503 if read err and file is existed #1992

### PR DESCRIPTION
**Pull request by jizhuozhi**
I hereby granted the copyright of the changes in this pull request
to the authors of this lua-nginx-module project.

### CODE REVIEW

**Comment by zhuizhuhaomeng** [comment]
I think 503 is better since content_by_lua_file will execute the Lua script in the file but not return the file itself.

**Comment by jizhuozhi** [comment]
> I think 503 is better since content_by_lua_file will execute the Lua script in the file but not return the file itself.

LGTM, this may be an appropriate semantic.

File: t/002-content.t [hunk 1]
```diff
@@ -850,3 +852,13 @@ GET /lua
 --- error_code: 500
 --- error_log eval
 qr/failed to load inlined Lua code: /
+
```
Comment (membphis) [inline t/002-content.t:855]: style: three blank lines between different test cases

**Comment by jizhuozhi** [comment]
Maybe we need a CONTRIBUTING GUIDE, there are too many code style problems when submitting code and writing test cases

**Comment by xiaocang** [comment]
> Maybe we need a CONTRIBUTING GUIDE, there are too many code style problems when submitting code and writing test cases

Here's the coding style of c: http://openresty.org/en/c-coding-style-guide.html
This is the tool to check `.t` file: https://github.com/openresty/openresty-devel-utils/blob/master/reindex

File: t/002-content.t [hunk 2]
```diff
@@ -1,5 +1,7 @@
 # vim:set ft= ts=4 sw=4 et fdm=marker:
 
+use strict;
+use warnings FATAL => 'all';
```
Comment (xiaocang) [inline t/002-content.t:4]: Here is the duplicate code, because the following `use Test::Nginx::Socket::Lua` will introduces the same restriction

**Review by xiaocang — CHANGES_REQUESTED** [review]
LGTM

File: t/002-content.t (hunk 2 above)
Comment (jizhuozhi) [reply t/002-content.t:4]: Sorry, it's CLion auto completed, I will remove it

**Comment by zhuizhuhaomeng** [comment]
@jizhuozhi  would you please change 503 to 500(Internal Server Error).

the 503 error:
```
The HyperText Transfer Protocol (HTTP) 503 Service Unavailable server error response code indicates that the server is not ready to handle the request.

Common causes are a server that is down for maintenance or that is overloaded. This response should be used for temporary conditions and the Retry-After HTTP header should, if possible, contain the estimated time for the recovery of the service.
```

**Comment by jizhuozhi** [comment]
> @jizhuozhi would you please change 503 to 500(Internal Server Error).
> 
> the 503 error:
> 
> ```
> The HyperText Transfer Protocol (HTTP) 503 Service Unavailable server error response code indicates that the server is not ready to handle the request.
> 
> Common causes are a server that is down for maintenance or that is overloaded. This response should be used for temporary conditions and the Retry-After HTTP header should, if possible, contain the estimated time for the recovery of the service.
> ```

I would not change the response code from `503` to `500`, cause as a proxy server, we need and must distinguish between proxy server errors and upstream business system errors.

**Comment by jizhuozhi** [comment]
@zhuizhuhaomeng And I think it is a special case of maintenance, cause file permission almost never changed on the fly without system admins. But for some careless admins, they might set wrong permissions

**Comment by jizhuozhi** [comment]
@xiaocang Hello, I'm sorry too long to remember resolving the conflict, PTAL.

**Review by zhuizhuhaomeng — APPROVED** [review]

**Pull request merged**